libopenage/engine/engine.cpp

```diff
@@ -53,31 +54,40 @@ Engine::Engine(mode mode,
 		this->time_loop.reset();
 	});
 
-	// if presenter is used, run it in a separate thread
+	log::log(INFO << "Using " << this->threads.size() + 1 << " threads "
+	              << "(" << std::jthread::hardware_concurrency() << " available)");
+}
+
+void Engine::loop() {
```

Comment:
The structure with `loop()` and `loop_simulation()` is a bit too convoluted now I think. I recommend we organize this in a different way:

1. Move all thread functions (time loop, simulation loop, presenter loop) into a separate function definition (maybe even outside `Engine` class?)
2. Move the thread spawning to `Engine::loop()` (so that alll threads are started there)
3. In `Engine::loop()` it should be much clearer why we are running different functions in the main thread depending on headless mode. The comments should explain that this is because of the Qt requirements.